CRM/Report/Form.php

```diff
@@ -4957,4 +4959,22 @@ public function getPermissionedFTClauseForLineItem(&$clauses, $alias = NULL) {
     $clauses[] = $alias . '.' . $clause;
   }
 
+  /**
+   * Function to build select where clause for financial type ACL.
+   *
+   * @param object $query
+   *
+   */
+  public function getSelectWhereClause(&$query) {
+    $clauses = array();
+    $aliases = $query->getVar('_aliases');
+    CRM_Utils_Hook::selectWhereClause('aclReport', $clauses);
+    if (!empty($clauses)) {
+      foreach ($clauses as $field => $vals) {
+        $tableAlias = $aliases[$field];
```

Comment:
@eileenmcnaughton few thoughts on this.

I debugged the two functions above and tried to make use of mergeSubquery to add my condition but was still unable to get the aliases in. My thoughts are that we may need to modify getSelectWhereClause to accept an array of aliases and then let the function do the hook invocation. So instead of invoking the hook directly in Report/Form.php, we will have to call getSelectWhereClause and pass in our aliases. This will in turn provide the aliases to the hook and then we can make use of it.

Another question, perhaps unrelated, is about the use of hook_civicrm_alterReportVar. Is there any specific reason as to why we are not making use of this in reports? Or is it just because we want to shift to using this hook rather than the one which is already present.